jvm-packages/xgboost4j/src/native/xgboost4j.cpp

```diff
@@ -23,7 +24,11 @@
 // helper functions
 // set handle
 void setHandle(JNIEnv *jenv, jlongArray jhandle, void* handle) {
+#ifdef __APPLE__
   long out = (long) handle;
+#else
+  int64_t out = (int64_t) handle;
```

Comment:
int64_t is valid in gcc, but on a Mac it's defined to be a different type to a jlong (unlike Windows or Linux) so that's why the condition is on __APPLE__. They all end up being 64 bit integers, but on a Mac int64_t is long long int and jlong is long int.